app/assets/templates/header.html

```diff
@@ -5,14 +5,14 @@
       <div class="header-left col-xs-10">
         <div class="row">
           <a href="#/" class="lineup btn btn-default col-xs-1" id="logo"><img src="assets/logo.gif"  height="20px"></a>
-          <input type="text" id="search-bar" class="btn btn-default lineup col-xs-9" placeholder="Search">
+          <input ng-model="searchTerms" ng-keyup="search()" type="text" id="search-bar" class="btn btn-default lineup col-xs-9" placeholder="Search">
           <button id="dropdown" class="lineup btn btn-default dropdown-toggle col-xs-1" data-toggle="dropdown">
             <div class="glyphicon glyphicon-list"></div>
             <span class="sr-only">Toggle Dropdown</span>
           </button>
           <ul class="dropdown-menu" role="menu">
             <li ng-repeat='category in categories'>
-              <a href="#">{{ category.name }}</a>
+              <a data-id="{{category.id}}" ng-click="setCategory(category.id)" href="#{{category.link}}">{{ category.name }}</a>
```

Comment:
```data-id``` and ```ng-click`` are probably not needed anymore. Try using ```ng-href``` instead of ```href``` (cf. [ngHref](https://docs.angularjs.org/api/ng/directive/ngHref))